pkg/encrypt/server-side.go

```diff
@@ -101,6 +101,21 @@ func NewSSEC(key []byte) (ServerSide, error) {
 	return sse, nil
 }
 
+// SSE transforms a SSE-C copy encryption into a SSE-C encryption.
+// It is the inverse of SSECopy(...).
+//
+// If the provided sse is no SSE-C copy encryption SSE returns
+// sse unmodified.
+func SSE(sse ServerSide) ServerSide {
```

Comment:
SSE() should be called as SSEC() here i think, since it might be confusing when we implement NewSSE()

Reply:
`NewSSE()` is already [there](https://github.com/minio/minio-go/blob/master/pkg/encrypt/server-side.go#L97) - as well as the `NewSSEC()` [function](https://github.com/minio/minio-go/blob/master/pkg/encrypt/server-side.go#L113)

Semantically the current solution is sound:
`encrypt.SSE(encrypt.NewSSE())` is equivalent to `encrypt.NewSSE()`
`encrypt.SSE(encrypt.DefaultPBKDF(...))` is equivalent to `encrypt.DefaultPBKDF(...)`
`encrypt.SSE(encrypt.SSECopy(encrypt.NewSSE()))` is equivalent to `encrypt.NewSSE()`
`encrypt.SSE(nil)` is equivalent to `nil`
a.s.o.

`SSE()` and `SSECopy()` are "generic" functions. Specifying `copy` only for SSE-C may be correct with respect to functionality (only SSE-C distinguish copy / non-copy headers) but probably leads  to more complexity for the client because he/she has to understand all that.
Right now `encrypt.SSE(encrypt.SSE(encrypt.SSECopy(encrypt.NewSSE())))` boils down to `encrypt.SSE()` semantically. So I don't see how clients can get confused - except if they want to do advanced things **and** don't read the function doc.

Reply:
So you're saying a user could confuse the function call `encrypt.SSE(..)` with a type conversion:
`encrypt.SSE(...)` like `[]byte(...)`? (A struct initialization would require `{` brackets)
 
Semantically it's exactly that - but as a function. Another (maybe more explicit) version would
be `encrypt.ToSSE(...)` - however the API would look a bit odd with `SSECopy()` and `ToSSE()`.
Or we break compatibility and add to functions to the interface to convert back and forth.
(There should be no other implementations because there is only SSE-S3, SSE-C and SSE-KMS)

Reply:
okay fair enough I think its fine...